service/history/handler.go

```diff
@@ -206,6 +209,11 @@ func (h *Handler) DeepHealthCheck(
 		return &historyservice.DeepHealthCheckResponse{State: enumsspb.HEALTH_STATE_DECLINED_SERVING}, nil
 	}
 
+	rsp := h.checkHistoryHealthSignals()
+	if rsp != nil {
+		return rsp, nil
+	}
+
```

Comment:
Can you check if the L221 should be metrics.HistoryHostHealthGauge.With(h.metricsHandler).Record(float64(enumsspb.HEALTH_STATE_NOT_SERVING))

Reply:
oh you're correct -- it should be NotServing, not DeclinedServing. Changed, thanks!